File: 4.8-bc/ts/stage-3-hexagonal/src/notifications/domain/notification.ts
// Notifications Bounded Context.
// Notifications — sink BC: підписується на події інших BC і шле email/push/sms.

export type Channel = "email" | "push" | "sms";

export const CHANNEL_EMAIL: Channel = "email";
export const CHANNEL_PUSH: Channel = "push";
export const CHANNEL_SMS: Channel = "sms";

export interface Notification {
  id: string;
  userId: string;
  channel: Channel;
  payload: string;
  sentAt: Date | null;
}

// Sender — port. Реалізації живуть у notifications/infra/{email,push,stub}.
export interface Sender {
  send(n: Notification): Promise<void>;
}

File: 4.8-bc/go/stage-3-hexagonal/notifications/domain/notification.go
// Package domain — Notifications Bounded Context.
// Notifications — sink BC: підписується на події інших BC і шле email/push/sms.
package domain

import (
	"context"
	"time"

	"github.com/google/uuid"
)

type Channel string

const (
	ChannelEmail Channel = "email"
	ChannelPush  Channel = "push"
	ChannelSMS   Channel = "sms"
)

type Notification struct {
	ID      uuid.UUID
	UserID  uuid.UUID
	Channel Channel
	Payload string
	SentAt  *time.Time
}

// Sender — port. Реалізації живуть у notifications/infra/{email,push,stub}.
type Sender interface {
	Send(ctx context.Context, n Notification) error
}

type Repository interface {
	Save(ctx context.Context, n Notification) error
}
